test.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+
+FAILED_TESTS=""
+
+# restore the previous logger config file
+function clean_up {
+	rm ../../logger/logger.config
+	mv ../../logger/logger.config.orig ../../logger/logger.config
+}
+
+function run_one_test {
+	local found_test=0    # whether or not we found the specified test
+	local status=0        # whether or not the test passed
+	local test_exe=""     # name of executable for test
+
+	# search through all the tests to find the source file corresponding to specified test
+	for test in $(ls); do
+		# skip Makefile and anything that doesn't match what was specified
+		if [[ $test == "Makefile" || $test != *"$1.c" ]]; then
+			continue
+		fi
+		found_test=1
+		break
+	done
+
+	# exit if we didn't find the test
+	if [[ $found_test == 0 ]]; then
+		printf "Could not find specified test $1\n"
+		clean_up
+		exit 1
+	fi
+
+	# make executable (remove the executable and re-make if it exists)
+	test_exe=$(echo $test | awk -F'.' '{ print $1 }')
+	if [[ -f $test_exe ]]; then
+		rm $test_exe
+	fi
+	make $test_exe
+
+	# run test
+	printf "\nRunning $test...\n"
+	./$test_exe
+	status=$?
+
+	# move the original log file back
+	clean_up
+
+	# exit with error status of test
+	if [[ $status == 1 ]]; then
+		exit 1
+	fi
+	exit 0
+}
+
+function run_all_tests {
+	local all_is_well=1 # is set to 0 if we have any failed tests
+	local status=0      # whether or not the test pased
+
+	# remove and recompile all tests
+	make clean
+	make
+
+	# loop through all tests
+	for test in $(ls); do
+	    # skip source files and Makefile
+	    if [[ $test == "Makefile" || $test == *".c" ]]; then
+	        continue
+	    fi
+
+		# run test
+		printf "\nRunning $test...\n"
+		./$test
+		status=$?
+
+		# if that test failed, set all_is_well to 1
+		if [[ $status == 1 ]]; then
+			FAILING_TESTS="$FAILING_TESTS $test"  # add this test to list of failing testse
+			all_is_well=0
+		fi
+	done
+
+	# move the original log file back
+	clean_up
+
+	# return status
+	if [[ $all_is_well == 0 ]]; then
+		printf "\n\n################################################# TESTS FAILED! ##########################################\n\n"
+		printf "Failing tests: $FAILING_TESTS\n\n\n"
+		exit 1
+	fi
+}
+
+################################################ BEGIN SCRIPT ##########################################
+
+# build all of Runtime
+./build.sh
+
+# do some setup work with the log file
+mv logger/logger.config logger/logger.config.orig
+cp -p tests/logger/logger.config logger/logger.config
+
+export PYTHONPATH="../tests/student_code" # this is so that executor can find the sample student code we write for testing
```

Comment:
They're preferred, but then that would make it so that the tests can only be run on the `pi` when the directory structure is `/home/pi/c-runtime`, for example. Will it break on Travis if that's not the absolute path of the `student_code` directory?

Reply:
I actually fixed this on my end. You can always use $PWD to get the current directory to construct the absolute path

Reply:
Oh true, I forgot about that. Will add~~